en/lessons/libraries/guardian.md

```diff
@@ -83,33 +83,31 @@ end
 `config/config.ex`
 
 ```elixir
-# in each environment config file you should overwrite this if it's external
-config :guardian, Guardian,
+config :guardian, MyApp.Guardian,
   issuer: "MyAppId",
-  secret_key: Mix.env(),
-  serializer: MyApp.GuardianSerializer
+  secret_key: Mix.env, # in each environment config file you should overwrite this if it's external
 ```
 
-This is the minimum set of information you need to provide Guardian with to operate. You shouldn't encode your secret key directly into your top-level config. Instead, each environment should have its own key. It's common to use the Mix environment for secrets in dev and test. Staging and production, however, must use strong secrets. (e.g. generated with `mix phoenix.gen.secret`)
+This is the minimum set of information you need to provide Guardian with to operate. You shouldn't encode your secret key directly into your top-level config. Instead, each environment should have its own key. It's common to use the Mix environment for secrets in dev and test. Staging and production, however, must use strong secrets. (e.g. generated with `mix guardian.gen.secret`)
 
-`lib/my_app/guardian_serializer.ex`
+`lib/my_app/guardian.ex`
 
 ```elixir
-defmodule MyApp.GuardianSerializer do
-  @behaviour Guardian.Serializer
+defmodule MyApp.Guardian do
+  use Guardian, otp_app: :my_app
 
   alias MyApp.Repo
   alias MyApp.User
 
-  def for_token(user = %User{}), do: {:ok, "User:#{user.id}"}
-  def for_token(_), do: {:error, "Unknown resource type"}
+  def subject_for_token(user = %User{}, _claims), do: { :ok, "User:#{user.id}" }
+  def subject_for_token(_, _), do: { :error, "Unknown resource type" }
 
-  def from_token("User:" <> id), do: {:ok, Repo.get(User, id)}
-  def from_token(_), do: {:error, "Unknown resource type"}
+  def resource_from_claims(%{"sub" => "User:" <> id}), do: { :ok, Repo.get(User, id) }
+  def resource_from_claims(_claims), do: { :error, "Unknown resource type" }
 end
 ```
-Your serializer is responsible for finding the resource identified in the `sub` (subject) field. This could be a lookup from a db, an API, or even a simple string.
-It's also responsible for serializing a resource into the `sub` field.
+
+Your implementation module is responsible for finding the resource identified in the `sub` (subject) field. This could be a lookup from a db, an API, or even a simple string. Serializing your resource into the `sub` field and provides many callbacks to interact with a token throughout it's lifecycle which we'll get to later.
```

Comment:
It seems like the third sentence here needs to be rephrased.

Reply:
Thanks. Updated